### PR TITLE
docs(demo): remake the whole style based on UX recommandations

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

Implemented the new look of the Vapor demo based on UX mockup https://coveo.invisionapp.com/share/7YWER75UCEJ#/409101170_Components

### Potential Breaking Changes

None.

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by oliviertaillon** [comment]
> I don't know if we want to improve the layout on smaller screen, but on half my monitor the content is quite small
> image

We could add a min-width of ~800px to #App

**Comment by gdostie** [comment]
The old vapor demo page also had this problem so its out of scope of this change.

image

Since this is still a problem which is quite easily fixable with css media queries, I entered a story in our internal backlog.

File: packages/demo/src/components/index.tsx [hunk 1]
```diff
@@ -49,15 +49,17 @@ const Components: React.FunctionComponent<RouteComponentProps> = ({match}) => {
     }
 
     return (
-        <div className="coveo-form flex full-content">
+        // <div className="coveo-form flex full-content">
+        <>
             <SideMenu components={components} />
-            <div className="flex-auto overflow-auto relative">
+            <div className="coveo-form flex-auto relative bg-pure-white shadow-3 m4 overflow-auto">
```
[the remark this answers is not in the document]
Comment (gdostie) [reply packages/demo/src/components/index.tsx]: We could in the components section, but not in the styles section so the size of the content box would differ between the 2 sections

[the remark this answers is not in the document]
Comment (gdostie) [reply packages/demo/src/components/index.tsx]: Had to do some refactoring to re-add the side nav scroll bar. I was able to remove the margin in both sections.

**Pull request merged**